src/tiled/objectselectiontool.cpp

```diff
@@ -1431,20 +1489,43 @@ void ObjectSelectionTool::saveSelectionState()
 void ObjectSelectionTool::refreshCursor()
 {
     Qt::CursorShape cursorShape = Qt::ArrowCursor;
-
+    QHash<MapObject*, MapObjectHighlight*> highlightItems;
```

Comment:
> I think especially for the second case, it could make sense to move the management of this hover highlight into ObjectSelectionItem, because it is already connected to all the relevant signals. 

Yes, I'll start working on this change. Ran into some class assignments lately, so I may need a day or two.

> would be necessary to communicate the hovered object from the ObjectSelectionTool to the ObjectSelectionItem. I think it would be best to do this through the MapDocument, adding an mHoveredObject pointer there.

How will this exactly solve dereferencing a roaming pointer? Or you mean to say since the second issue will be solved, this issue can not be reproduced?